tests/system/feature_store/test_feature_store.py

```diff
@@ -367,6 +368,35 @@ def test_different_target_paths_for_get_offline_features(self, target_path):
         assert df is not None
         assert target.run_id == "offline-features"
 
+    @pytest.mark.parametrize(
+        "raise_error, pass_run_id, is_single_file, target_path",
+        [
+            (False, True, False, "v3io:///bigdata/{run_id}"),
+            (False, True, True, "v3io:///bigdata/{run_id}/file.parquet"),
+            (False, False, False, "v3io:///bigdata/"),
+            (False, False, True, "v3io:///bigdata/file.parquet"),
+            (False, True, False, "v3io:///bigdata/"),
+            (False, True, True, "v3io:///bigdata/file.parquet"),
+            (True, False, False, "v3io:///bigdata/{run_id}"),
+            (True, False, True, "v3io:///bigdata/{run_id}/file.parquet"),
+        ],
+    )
+    def test_different_target_path_scenarios_for_run_id(
+        self, raise_error, pass_run_id, is_single_file, target_path
+    ):
+        run_id = "run_id_val" if pass_run_id else None
+
+        if raise_error:
+            with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+                TargetPathObject(target_path, run_id, is_single_file)
+        else:
+            tp_obj = TargetPathObject(target_path, run_id, is_single_file)
+            assert (mlrun.model.RUN_ID_PLACE_HOLDER in tp_obj.get_templated_path()) == (
+                (pass_run_id and not is_single_file)
+                or mlrun.model.RUN_ID_PLACE_HOLDER in target_path
+            )
+            assert mlrun.model.RUN_ID_PLACE_HOLDER not in tp_obj.get_absolute_path()
```

Comment:
Maybe I'm missing something but it doesn't seem like you're actually doing something here that require a system, therefore no reason for this to be a system test
Move to `tests/test_model.py`